### PR TITLE
feature/add config factory

### PR DESCRIPTION
**Pull request by jamescallumyoung**
- Add partial-builder factory - add interfaces for the eslint config file and the library config files - add factory for creating eslint configs from library configs

### CODE REVIEW

**Review by giovannifilannino — APPROVED** [review]
LGTM

File: src/lib/interfaces/eslint-config-partial.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface EslintConfigPartial {
```
Comment (vabatta) [inline src/lib/interfaces/eslint-config-partial.ts:1]: This should be renamed as definition file `.d.ts`.

File: src/lib/interfaces/library-config.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+interface OptionalExport {
```
Comment (vabatta) [inline src/lib/interfaces/library-config.ts:1]: This should be renamed as definition file `.d.ts`.

**Review by vabatta — CHANGES_REQUESTED** [review]

**Review by giovannifilannino — DISMISSED** [review]

**Comment by jamescallumyoung** [comment]
Added a jest test file to test the behaviour. The tests don't auto-run on the branches since we have no CI/CD so if you want to test you'll have to check it out and test it.

(try: `gh pr checkout 16 && yarn && yarn run test`)

**Review by vabatta — APPROVED** [review]

**Pull request merged**